src/api/cvc4cpp.cpp

```diff
@@ -90,5 +91,571 @@ std::ostream& operator<<(std::ostream& out, const Result& r)
+        {CVC4::Kind::STRING_CONCAT, STRING_CONCAT},
+        {CVC4::Kind::STRING_IN_REGEXP, STRING_IN_REGEXP},
+        {CVC4::Kind::STRING_LENGTH, STRING_LENGTH},
+        {CVC4::Kind::STRING_SUBSTR, STRING_SUBSTR},
+        {CVC4::Kind::STRING_CHARAT, STRING_CHARAT},
+        {CVC4::Kind::STRING_STRCTN, STRING_STRCTN},
+        {CVC4::Kind::STRING_STRIDOF, STRING_STRIDOF},
+        {CVC4::Kind::STRING_STRREPL, STRING_STRREPL},
+        {CVC4::Kind::STRING_PREFIX, STRING_PREFIX},
+        {CVC4::Kind::STRING_SUFFIX, STRING_SUFFIX},
+        {CVC4::Kind::STRING_ITOS, STRING_ITOS},
+        {CVC4::Kind::STRING_STOI, STRING_STOI},
+        {CVC4::Kind::CONST_STRING, CONST_STRING},
+        {CVC4::Kind::STRING_TO_REGEXP, STRING_TO_REGEXP},
+        {CVC4::Kind::REGEXP_CONCAT, REGEXP_CONCAT},
+        {CVC4::Kind::REGEXP_UNION, REGEXP_UNION},
+        {CVC4::Kind::REGEXP_INTER, REGEXP_INTER},
+        {CVC4::Kind::REGEXP_STAR, REGEXP_STAR},
+        {CVC4::Kind::REGEXP_PLUS, REGEXP_PLUS},
+        {CVC4::Kind::REGEXP_OPT, REGEXP_OPT},
+        {CVC4::Kind::REGEXP_RANGE, REGEXP_RANGE},
+        {CVC4::Kind::REGEXP_LOOP, REGEXP_LOOP},
+        {CVC4::Kind::REGEXP_EMPTY, REGEXP_EMPTY},
+        {CVC4::Kind::REGEXP_SIGMA, REGEXP_SIGMA},
+        {CVC4::Kind::REGEXP_RV, INTERNAL_KIND},
+        /* Quantifiers ----------------------------------------------------- */
+        {CVC4::Kind::FORALL, FORALL},
+        {CVC4::Kind::EXISTS, EXISTS},
+        {CVC4::Kind::INST_CONSTANT, INTERNAL_KIND},
+        {CVC4::Kind::INST_PATTERN, INTERNAL_KIND},
+        {CVC4::Kind::BOUND_VAR_LIST, INTERNAL_KIND},
+        {CVC4::Kind::INST_NO_PATTERN, INTERNAL_KIND},
+        {CVC4::Kind::INST_ATTRIBUTE, INTERNAL_KIND},
+        {CVC4::Kind::INST_PATTERN_LIST, INTERNAL_KIND},
+        {CVC4::Kind::INST_CLOSURE, INTERNAL_KIND},
+        {CVC4::Kind::REWRITE_RULE, INTERNAL_KIND},
+        {CVC4::Kind::RR_REWRITE, INTERNAL_KIND},
+        {CVC4::Kind::RR_REDUCTION, INTERNAL_KIND},
+        {CVC4::Kind::RR_DEDUCTION, INTERNAL_KIND},
+        /* Sort kinds ------------------------------------------------------ */
+        {CVC4::Kind::ARRAY_TYPE, INTERNAL_KIND},
+        {CVC4::Kind::ASCRIPTION_TYPE, INTERNAL_KIND},
+        {CVC4::Kind::CONSTRUCTOR_TYPE, INTERNAL_KIND},
+        {CVC4::Kind::DATATYPE_TYPE, INTERNAL_KIND},
+        {CVC4::Kind::SELECTOR_TYPE, INTERNAL_KIND},
+        {CVC4::Kind::SET_TYPE, INTERNAL_KIND},
+        {CVC4::Kind::SORT_TAG, INTERNAL_KIND},
+        {CVC4::Kind::SORT_TYPE, INTERNAL_KIND},
+        {CVC4::Kind::TESTER_TYPE, INTERNAL_KIND},
+        {CVC4::Kind::TYPE_CONSTANT, INTERNAL_KIND},
+        {CVC4::Kind::FUNCTION_TYPE, INTERNAL_KIND},
+        {CVC4::Kind::SEXPR_TYPE, INTERNAL_KIND},
+        {CVC4::Kind::BITVECTOR_TYPE, INTERNAL_KIND},
+        {CVC4::Kind::FLOATINGPOINT_TYPE, INTERNAL_KIND},
+        /* ----------------------------------------------------------------- */
+        {CVC4::Kind::LAST_KIND, LAST_KIND},
+    };
```

Comment:
Should we maybe add a function that converts internal to external kinds, which maps to `INERNAL_KIND` by default if the external kind is not in the map? This would get rid of all the `INTERNAL_KIND` mappings and does not force us to add internal only kinds to the map.

Reply:
Yeah, I was thinking about that, too. I guess it is a matter of performance at the end of the day.